engine/api/workflow/execute_node_job_run.go

```diff
@@ -316,222 +312,61 @@ func checkStatusWaiting(ctx context.Context, store cache.Store, jobID int64, sta
 	return nil
 }
 
-// LoadNodeJobRunKeys loads all keys for a job run
-func LoadNodeJobRunKeys(ctx context.Context, db gorp.SqlExecutor, proj *sdk.Project, wr *sdk.WorkflowRun, nodeRun *sdk.WorkflowNodeRun) ([]sdk.Parameter, []sdk.Variable, error) {
-	var app *sdk.Application
-	var env *sdk.Environment
-
-	n := wr.Workflow.WorkflowData.NodeByID(nodeRun.WorkflowNodeID)
-	if n.Context.ApplicationID != 0 {
-		appMap, has := wr.Workflow.Applications[n.Context.ApplicationID]
-		if has {
-			app = &appMap
-			keys, err := application.LoadAllKeysWithPrivateContent(db, app.ID)
-			if err != nil {
-				return nil, nil, err
-			}
-			app.Keys = keys
-		}
-	}
-	if n.Context.EnvironmentID != 0 {
-		envMap, has := wr.Workflow.Environments[n.Context.EnvironmentID]
-		if has {
-			env = &envMap
-			keys, err := environment.LoadAllKeysWithPrivateContent(db, n.Context.EnvironmentID)
-			if err != nil {
-				return nil, nil, err
-			}
-			env.Keys = keys
-		}
+// LoadSecrets loads all secrets for a job run
+func LoadSecrets(ctx context.Context, db gorp.SqlExecutor, wr *sdk.WorkflowRun, nodeRun *sdk.WorkflowNodeRun) ([]sdk.Variable, error) {
+	secrets := make([]sdk.Variable, 0)
+
+	projetSecrets, err := loadRunSecretByContext(ctx, db, wr.ID, "proj")
```

Comment:
It could be nice to make only one request

Reply:
done